lib/papercups_flutter.dart

```diff
@@ -155,20 +135,14 @@ class _PaperCupsWidgetState extends State<PaperCupsWidget> {
   }
 
   @override
-  void didUpdateWidget(PaperCupsWidget oldWidget) {
+  void didUpdateWidget(PapercupsWidget oldWidget) {
     super.didUpdateWidget(oldWidget);
-    if (oldWidget.props.primaryColor != widget.props.primaryColor) {
-      if ((widget.props.primaryColor != null &&
-              widget.props.primaryColor!.computeLuminance() > 0.5) ||
-          (widget.props.primaryGradient != null &&
-              widget.props.primaryGradient!.colors[0].computeLuminance() >
-                  0.5) ||
-          (widget.props.primaryColor == null &&
-              Theme.of(context).primaryColor.computeLuminance() > 0.5)) {
+    if (oldWidget.props.style.primaryColor != widget.props.style.primaryColor) {
```

Comment:
Maybe we want to optionally disable computeLuminance if we're already setting everything else up in `PapercupsStyle`?

Reply:
I think it'd make sense to keep it if some people set `primaryColor`/`primaryGradient`, but not `userBubbleTextStyle `/`userAttachmentTextStyle`. By keeping the `computeLuminance` the latter 2 - which use `widget.textColor` -  could then have their luminance be automatically adapted to the color the user provided.